client/src/pages/locations/Form.js

```diff
@@ -138,16 +140,36 @@ const BaseLocationForm = ({ currentUser, edit, title, initialValues }) => {
                   name="location"
                   component={FieldHelper.ReadonlyField}
                   humanValue={
-                    <>
-                      <Coordinate coord={values.lat} />,{" "}
-                      <Coordinate coord={values.lng} />
-                    </>
+                    <GeoLocation
+                      lat={values.lat}
+                      lng={values.lng}
+                      isSubmitting={isSubmitting}
+                      onClearLocation={() => onClearLocation(setFieldValue)}
```

Comment:
If `GeoLocation` is mostly a read-only component, it is a bit strange that it does the clearing of the location. I suggest to have a location editor (you can evolve GeoLocation into that) that does the following:
- it takes a location data model consisting of a `lat, lon`
- it can edit that data model in a certain format, including the ability to clear it (this way no need to pass a onClearLocation)
- it can be configured in the future to display in different formats (MGRS, etc)
- it reuses some code between a read-only renderer and an editor